cmd/integration_test.go

```diff
@@ -395,19 +441,42 @@ func TestSmokescreenIntegration(t *testing.T) {
 			conformResult(t, testCase, resp, err, logHook.AllEntries())
 		})
 	}
+}
+
+// This test must be run with a separate test command as the environment variables
+// required can race with the test above.
+func TestUpstreamProxySmokescreenIntegration(t *testing.T) {
```

Comment:
This just tests that ill-formed proxy configurations elicit meaningful error messages, right? If so, `TestInvalidProxyConfiguration` or similar would be more descriptive.